dialogue-apache-hc5-client/src/main/java/com/palantir/dialogue/hc5/InstrumentedPoolingHttpClientConnectionManager.java

```diff
@@ -58,12 +60,41 @@
 
     @Override
     public void close() {
-        manager.close();
+        if (!closed) {
+            log.warn(
+                    "Dialogue ConnectionManager close invoked unexpectedly and ignored",
+                    SafeArg.of("clientName", clientName),
+                    new SafeRuntimeException("stacktrace"));
+            // Note: manager.close is not invoked here, see closeUnderlyingConnectionManager.
+        }
```

Comment:
Relevant upstream link is: https://github.com/apache/httpcomponents-client/blob/5b61e132c3871ddfa967ab21b3af5d6d738bc6e8/httpclient5/src/main/java/org/apache/hc/client5/http/impl/classic/MainClientExec.java#L161-L164